Roc pc plus

cljidkx opened the pull request:
Add new device RK3399-ROC-PC-PLUS, and modify ES8328 to support ES8388 chip.  Does this modification meet your rules?

chewitt commented:
@cljidkx Adding support for new devices to LE is a two stage process:

Step 1 = Submit device-tree and driver change patches to the Linux kernel, respond to feedback and get them merged or queued for a future kernel release.

Step 2 = Once patches are accepted upstream, open a PR to add the board to our codebase and we will be happy to add the required backported patches.

Please understand that we are not interested in adding support for boards that are not upstream, because then we are stuck with "out of tree" patches forever. If the changes are accepted upstream we know they can dropped in a future kernel bump and there is no maintenance burden on our project. NB: We do accept "work in progress" patches, but only from frequent contributors or staff maintainers who we know and trust to upstream the changes in parallel. For everyone else, Step 1 is not negotiable.

If you need guidance on upstreaming please ask via forums or email and we will be happy to coach.

knaerzche commented:
@cljidkx: I really don't want to anticipate review from kernel maintainers, but some general remark: Its in general a bad idea to compare to devicetrees in vendor kernel and drop/add changes from there even if they might look similar. Drivers in upstream kernel are different in almost all cases and require different device tree entries. And: you really should not add changes to dtsi files which are used by other boards, just because you need it for yours: If its really required: change it in board's dts.

cljidkx commented:
> @cljidkx Adding support for new devices to LE is a two stage process:
> 
> Step 1 = Submit device-tree and driver change patches to the Linux kernel, respond to feedback and get them merged or queued for a future kernel release.
> 
> Step 2 = Once patches are accepted upstream, open a PR to add the board to our codebase and we will be happy to add the required backported patches.
> 
> Please understand that we are not interested in adding support for boards that are not upstream, because then we are stuck with "out of tree" patches forever. If the changes are accepted upstream we know they can dropped in a future kernel bump and there is no maintenance burden on our project. NB: We do accept "work in progress" patches, but only from frequent contributors or staff maintainers who we know and trust to upstream the changes in parallel. For everyone else, Step 1 is not negotiable.
> 
> If you need guidance on upstreaming please ask via forums or email and we will be happy to coach.

Does such a change have a chance to merge?

chewitt commented:
The best way to find out what the Rockchip kernel maintainers think about your changes is to send the patches to their mailing list tagged with RFC .. and let them comment. Their main interest is that your changes a) work, b) don't break something else, c) follow kernel submission (style) rules. Your current changes have numerous style issues, but if you explain this is your first submission to the kernel they will focus on the content of the changes not the presentation (for now). Some of the maintainers also lurk in the #linux-rockchip IRC channel on Libera.Chat, so perhaps ask for advice there too.

knaerzche commented:
@cljidkx Maybe a little motivation on submitting to upstream: I just submitted the node/s for the ir-receiver for the common rk3399-roc-pc.dtsi (for use with the generic driver) just a few days ago: https://patchwork.kernel.org/project/linux-rockchip/patch/20210619121642.7892-1-knaerzche@gmail.com/ and it will now land in kernel 5.14.
Its really not hard for those super simple device tree patches: A very detailed guide can be found here: https://www.kernel.org/doc/html/v5.12/process/submitting-patches.html. A super simplified guide (which should cover your case) can be found here: http://nickdesaulniers.github.io/blog/2017/05/16/submitting-your-first-patch-to-the-linux-kernel-and-responding-to-feedback/

cljidkx commented:
> @cljidkx Maybe a little motivation on submitting to upstream: I just submitted the node/s for the ir-receiver for the common rk3399-roc-pc.dtsi (for use with the generic driver) just a few days ago: https://patchwork.kernel.org/project/linux-rockchip/patch/20210619121642.7892-1-knaerzche@gmail.com/ and it will now land in kernel 5.14.
> Its really not hard for those super simple device tree patches: A very detailed guide can be found here: https://www.kernel.org/doc/html/v5.12/process/submitting-patches.html. A super simplified guide (which should cover your case) can be found here: http://nickdesaulniers.github.io/blog/2017/05/16/submitting-your-first-patch-to-the-linux-kernel-and-responding-to-feedback/

Yes, you are right.  I have now deleted the ir device and modified some kernel patch style issues

chewitt commented:
```
git fetch upstream/master
git rebase upstream/master
git push -f origin roc-pc-plus
```
^ this will rebase your changes against our master branch and push to origin on github, to drop the merge commits from the PR. There are a number of missing line-space and double line-space style errors in the dts files. It's the kind of style thing that maintainers will want corrected.

knaerzche commented:
**Like already told (maybe its still not clear yet): Please submit your patches to the upstream kernel, they will not get accepted otherwise here in LE.** 

After a quick peak & to avoid being frustrated after the first answer from kernel maintainer: 
- does not exist mainline: https://github.com/LibreELEC/LibreELEC.tv/blob/9097191db6e06b9a774aa247e8e2adb5ac9c7d52/projects/Rockchip/patches/linux/default/linux-3000-rk3399-roc-pc-plus-adaptation.patch#L75-L98 - drop it
- since the wifi chipset you mentioned is a USB chipset sdio likely does not need to get enabled: https://github.com/LibreELEC/LibreELEC.tv/blob/9097191db6e06b9a774aa247e8e2adb5ac9c7d52/projects/Rockchip/patches/linux/default/linux-3000-rk3399-roc-pc-plus-adaptation.patch#L270-L283 - drop it
-  also drop the changes to rk3399.dtsi (as I've said already) - do it in your rk3399-roc-pc-plus.dts
- do not submit anything "commented out"
- pwm3 does not neeed to get enabled (since pins are driven by gpio-ir-receiver driver)
- maybe lots more
- check style (checkpatch.pl maybe tells someting)
- check ordering (order by name for nodes without "reg", order by "reg" otherwise (you don't have the latter))
-  check indentation
- overall: just compare to other board's devicetrees - that probably helps most
- run `make dtbs_check` - most driver's DT bindings are converted to yaml already so this is usually helpful also.
